lib/GitHub.js

```diff
@@ -182,7 +182,7 @@ class GitHub extends GitService {
     try {
       return await super.readFile(filePath, getFullResponse)
     } catch (err) {
-      return errorHandler('GITHUB_READING_FILE', {err})
+      throw errorHandler('GITHUB_READING_FILE', {err})
```

Comment:
This is the core of the fix. These async functions should throw for errors (will be transformed into a rejected `Promise`), not return an error object! The callers don't expect an error object so they do weird things like treat the error object as a file, etc.